geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/concurrent/CachedThreadPool.scala

```diff
@@ -86,10 +88,19 @@ class CachedThreadPool(maxThreads: Int) extends AbstractExecutorService {
     lock.lock()
     try {
       if (available > 0) {
+        // note that we could fairly easily create a global thread limit by backing this with a different pool
+        try { CachedThreadPool.pool.execute(task) } catch {
+          case e: RejectedExecutionException =>
+            // we still need to execute the task to fulfill the executor API
```

Comment:
Does the ExitingExecutor satisfy this part of the API contract?

Reply:
I think so - if a pool is 'shutdown' then it will reject any incoming execution requests, but it still has to run any tasks that were already submitted. if it's 'shutdownNow' then it can reject already submitted tasks. The issue we hit is that the static backing executor service can be shutdown by the jvm shutdown hook, which then causes us to not execute already submitted tasks.

Reply:
Ok, I think I see how the layers of this are fitting together.  The ExitingExecutor has a System shutdown hook registered.  The thing we build on top of that needs to be able to run one last task...  but the Executor is gonna already due to the System shutdown.  

Nice.